src/js/plugin/sdk/pipeline/PipelineModel.js

```diff
@@ -0,0 +1,48 @@
+
+class PipelineModel {
+
+  constructor() {
+    this.pipelines = new Map();
+  };
+
+  clearPipeline() {
+    this.pipelines.clear();
+  };
+
+  applyFilter(operationName, filterParameters) {
```

Comment:
I'd rename this method to `applyPipeline` as a pipeline could be used for many things other than filtering a result set otherwise I'd argue that we should restructure the model to prove a predicate that one can apply instead of applying the filter.

Reply:
:+1: